src/ert/config/analysis_config.py

```diff
@@ -29,36 +29,62 @@ class UpdateSettings:
     std_cutoff: float = 1e-6
     alpha: float = 3.0
     auto_scale_observations: List[ObservationGroups] = field(default_factory=list)
-    min_required_realizations: int = 2
 
 
+@dataclass
 class AnalysisConfig:
-    def __init__(
-        self,
-        alpha: float = 3.0,
-        std_cutoff: float = 1e-6,
-        max_runtime: int = 0,
-        min_realization: int = 0,
-        update_log_path: Union[str, Path] = "update_log",
-        analysis_set_var: Optional[List[Tuple[str, str, str]]] = None,
-        num_iterations: int = 1,
-    ) -> None:
-        self._max_runtime = max_runtime
-        self.minimum_required_realizations = min_realization
-        self._alpha = alpha
-        self._std_cutoff = std_cutoff
-        self._update_log_path = Path(update_log_path)
-        self._min_realization = min_realization
-        self.num_iterations = num_iterations
+    max_runtime: Optional[int] = None
+    minimum_required_realizations: int = 0
+    update_log_path: Union[str, Path] = "update_log"
+    es_module: ESSettings = field(default_factory=ESSettings)
+    ies_module: IESSettings = field(default_factory=IESSettings)
+    observation_settings: UpdateSettings = field(default_factory=UpdateSettings)
```

Comment:
Should we create an issue for not default initializing this?

Reply:
Can do, should also apply to the rest of the constructor as well probably